### PR TITLE
Update DVC training stages - use pre-extraced features

### LINKED ISSUE #3: Refactor the `train` and `eval` stages in DVC
After refactoring the training functionalities in order to have separate DVC stages for feature extraction, we need to adapt the `train` and `eval` stages to use the features produced by those previous stages.

### PR DESCRIPTION
**Pull request by Stannislav**
Closes #3 

A new command `morphoclass extract-features` was addede recently, and along with it a number of DVC stages named `features-*`. While the package has already been upated to use pre-extraced features in training, the DVC stages were still outdated. Therefore, the main goal of this PR is to update all `train-*` and `evaluate-*` DVC stages to use pre-extracted features.

Additional bug fixes:
* Fix a bug in `train_ml_model`: convert torch tensors to numpy array for training sklearn models
* Fix a bug in persistence diagram/image feature extraction: explicitly use dtype `float32` to be compatible with model weights.

### CODE REVIEW

**Comment by FrancescoCasalegno** [comment]
I guess this closes https://github.com/BlueBrain/morphoclass/issues/3?

**Comment by Stannislav** [comment]
> I guess this closes #3?

Ah yes. I updated the description.

**Comment by FrancescoCasalegno** [comment]
Two things that we may have to look into:

- [ ] Some files were modified after the last two commits f4f0d49 and 8fca841, so the `dvc.lock` changed. Should we re-trigger the `evaluation-*` stages?
- [ ] We are now fixing the `torch.load()` by specifying the `device`. However, running `dvc repro` on a CPU vs. GPU system would produce different artifacts. Should we also modify the code of `morphoclass training` to make sure that it's consistently written as CPU?
Basically I am talking about this:
https://github.com/BlueBrain/morphoclass/blob/8fca841b715af3057db4cbfde19773674e8668da/src/morphoclass/training/training_log.py#L177-L179
But I guess it's not trivial (maybe we need to iterate through each element of the `dict` and convert it?) because there's no equivalent of `torch.load`'s `map_location=device` for `torch.save`.

**Comment by FrancescoCasalegno** [comment]
## Training and Evaluation Results

There seems to be stages and corresponding results for `interneurons` (`in-*`) and `pyramidal-cells` (`pc-*`), but nothing for `janelia` (`janelia-*`).

- [ ] @Stannislav can you please add those?
